pkg/binder/factory.go

```diff
@@ -67,7 +69,10 @@ func DefaultUnitQueueSortFunc() framework.UnitLessFunc {
 func NewBasePlugins(victimsCheckingPlugins []*framework.VictimCheckingPluginCollectionSpec) *apis.BinderPluginCollection {
 	// TODO add some default plugins later
 	basicPlugins := apis.BinderPluginCollection{
-		CheckTopology: []string{},
+		CheckTopology: []string{
```

Comment:
nit: add these plugins on demand. We don't need to prepare date when the incoming Pod doesn't have cross-node constraints.

Reply:
Ok, I changed it to add `NodeTopologyCheckPlugins` as needed for each newTask at the beginning of the `CheckCrossNodeTopologyForUnit` function.

Reply:
After considering the `InterPodAffinity `constraint, the **antiAffinity of the existing pod** will affect the pod to be checked, so we need to check `InterPodAffinity`, and it is **difficult to determine in advance** whether `InterPodAffinity `is needed. The approach here is to check `InterPodAffinity `and `PodTopologySpread `by default, but in the check, we will check in advance whether the pod to be checked really has this constraint, and return quickly if not.